page-objects/src/components/AbstractElement.ts

```diff
@@ -22,8 +22,7 @@ export abstract class AbstractElement extends WebElement {
         let item: WebElement = AbstractElement.driver.findElement(By.css('html'));
         if (!enclosingItem) {
             enclosingItem = item;
-        }
-
+        }   
         if (enclosingItem instanceof WebElement) {
```

Comment:
This could potentially be 
```suggestion
        if (enclosingItem instanceof WebElement || enclosingItem instanceof ShadowRoot) {
```
since the enclosing element can now be either a WebElement or a ShadowRoot (which is the case for TextEditor's contextual menus

Reply:
@ajdor I am not sure about that. I am handling shadowRoot element before actually creating an instance. So I keep passing WebElement (from within shadowRoot context).